crates/agent/src/connector_tags.rs

```diff
@@ -145,12 +145,14 @@ impl TagHandler {
             #[serde(rename = "type")]
             protocol: String,
             resource_spec_schema: Box<RawValue>,
+            oauth2_spec: Option<Box<RawValue>>,
```

Comment:
@jgraettinger Good point. We already do have an OAuth2Spec in our flow protocol now (PR [here](https://github.com/estuary/flow/pull/570/files)), I will update this to deserialize and serialize again just to verify 👍🏽

Reply:
@jgraettinger I just tried this, and there seems to be a discrepancy between how proto types are serialized/deserialized to/from JSON between Go and Rust: in Go they are snake_case, while in Rust they are camelCase.

In Go we are serializing `flow.OAuth2Spec`, and it comes out with `snake_case` keys: https://github.com/estuary/flow/pull/570/files#diff-6f46f9561151d2babc68df1479eaef0ae774c36e2eb09a4902325adc0b08dc00R28

```
"oauth2Spec":{"provider":"google","auth_url_template":"https://accounts.google.com/o/oauth2/auth?access_type=offline\u0026client_id={{ client_id }}\u0026redirect_uri={{ redirect_uri }}\u0026response_type=code\u0026scope=email\u0026state={{ state }}","access_token_url_template":"https://oauth2.googleapis.com/token","access_token_body_json":{"grant_type":"authorization_code","client_id":"{{ client_id }}","client_secret":"{{ client_secret }}","redirect_uri":"{{ redirect_uri }}","code":"{{ code }}"}
```

However, in Rust, trying to deserialize this, I get:
```
Caused by:
    unknown field `auth_url_template`, expected one of `provider`, `authUrlTemplate`, `accessTokenUrlTemplate`, `accessTokenBody`, `accessTokenHeaders`, `refreshTokenUrlTemplate`, `refreshTokenBody`, `refreshTokenHeaders` at line 1 column 2414)
```

I can look into making these two compatible (i.e. both doing camelCase or both doing snake_case), but I'm afraid it might have a large blast radius. Any thoughts on this? I can also work around this by specifically parsing as snake_case in this instance in Rust and then serializing to camelCase before persisting it.

Reply:
Okay, I've explained here what is blocking me from verifying this: https://github.com/estuary/animated-carnival/issues/34

I think for now we can proceed with the raw JSON, but we should resolve that issue.